crates/provider/src/provider/trait.rs

```diff
@@ -1269,6 +1297,58 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    #[ignore]
+    async fn gets_tx_receipt_opt() {
+        init_tracing();
+        let url = "https://opt-mainnet.g.alchemy.com/v2/demo";
```

Comment:
we want avoid using live rpc providers because this can degrade CI drastically.

for these tests we should use the raw json objects and test deserialization like this.